Daemon API docs

VoidingWarranties opened the pull request:
Incomplete, see TODOs.

VoidingWarranties commented on doc/API.md:
```diff
@@ -82,104 +82,61 @@ Table of contents
+| [/daemon/constants](#daemonconstants-get) | GET       |
+| [/daemon/stop](#daemonstop-get)           | GET       |
+| [/daemon/version](#daemonversion-get)     | GET       |
 
```
Should have a note linking to `Wallet.md`. See Gateway for an example.

DavidVorick replied:
to wallet.md? Isn't this the daemon PR?

VoidingWarranties replied:
Ooops, right. I think I copy-pasted this from the other PR.

DavidVorick commented on doc/API.md:
```diff
@@ -82,104 +82,61 @@ Table of contents
+  "blocksizelimit":         2000000,    // bytes
+  "blockfrequency":         600,        // seconds per block
+  "targetwindow":           1000,       // blocks
+  "mediantimestampwindow":  11,         // TODO: What are the units? blocks?
```
blocks

DavidVorick commented on doc/API.md:
```diff
@@ -82,104 +82,61 @@ Table of contents
+  "siafundportion":         "39/1000",
+  "maturitydelay":          144,        // blocks
+
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
```
Siacoins. Definitely breaks from the convention of returning everything in hastings, but we have a compatibility promise that we need to uphold. So, I'd add some extra documentation explaining that, while this is presenting in siacoins and not hastings, it is unusual and doesn't follow the rest of the API.

DavidVorick commented on doc/API.md:
```diff
@@ -82,104 +82,61 @@ Table of contents
+  "maturitydelay":          144,        // blocks
+
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
+  "minimumcoinbase": 30000,  // TODO: See above TODO.
```
same deal

DavidVorick commented on doc/api/Daemon.md:
```diff
@@ -0,0 +1,96 @@
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
+  // Minimum number of coins paid out to the miner of a block (the coinbase
+  // decreases with each block).
+  "minimumcoinbase": 30000, // TODO: See above TODO.
```
:+1:

DavidVorick commented on doc/api/Daemon.md:
```diff
@@ -0,0 +1,96 @@
+  "maturitydelay": 144, // blocks
+
+  // Number of coins given to the miner of the first block.
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
```
:+1:

VoidingWarranties commented:
@DavidVorick this PR adds the "extremefuturetimestamp" field. In #1389 you mentioned adding a field for v1.1.0. Should we wait to add fields to the API until minor releases?

VoidingWarranties commented:
Rebased and updated. Good to go pending my above comment.

DavidVorick commented on doc/api/Daemon.md:
```diff
@@ -0,0 +1,100 @@
+  "mediantimestampwindow": 11, // blocks
+  // How far in the future a block can be without being rejected.
+  "futurethreshold": 10800, // seconds
+  // TODO: how is this different from futurethreshold?
```
TODO: document correctly :)

DavidVorick commented on doc/api/Daemon.md:
```diff
@@ -0,0 +1,100 @@
+
+  // Initial target.
+  "roottarget": [0,0,0,0,32,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
+  // Initial depth.
```
These feel incomplete, though I'm not sure how you'd expand the documentation here without getting too technical. Maybe 'Target hash for the first block following the genesis block' and 'Initial amount of work assigned to the blockchain.'

DavidVorick commented:
Yeah, we should wait to extend the API until 1.1, let's pull out the code and create an issue that we insert into milestone v1.1

VoidingWarranties commented:
Removed "extremefuturethreshold" field (to be added in another PR for v1.1.0), corrected "futurethreshold" comment, and rebased. Should be good to go.
